dotCMS/src/integration-test/java/com/dotcms/storage/FileMetadataAPITest.java

```diff
@@ -0,0 +1,370 @@
+package com.dotcms.storage;
+
+import static com.dotcms.datagen.TestDataUtils.FILE_ASSET_1;
+import static com.dotcms.datagen.TestDataUtils.FILE_ASSET_2;
+import static com.dotcms.datagen.TestDataUtils.FILE_ASSET_3;
+import static com.dotcms.datagen.TestDataUtils.getFileAssetContent;
+import static com.dotcms.datagen.TestDataUtils.getMultipleBinariesContent;
+import static com.dotcms.datagen.TestDataUtils.removeAnyMetadata;
+import static com.dotcms.storage.StoragePersistenceProvider.DEFAULT_STORAGE_TYPE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.datagen.TestDataUtils.TestFile;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableSet;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import io.vavr.Tuple2;
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FilenameUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class FileMetadataAPITest {
+
+    private static final String FILE_ASSET = "fileAsset";
+    private static FileMetadataAPI contentletMetadataAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)